src/Host/Protocol/Impl/Message.cs

```diff
@@ -36,37 +37,42 @@ public class Message {
             Blob = blob ?? new byte[0];
         }
 
-        public Message(byte[] data) {
+        private Message() { }
+
+        public static Message Create(byte[] data) {
```

Comment:
It was just a style choice. I did not want to put parsing code in the constructor.